ci_framework/roles/artifacts/tasks/main.yml

```diff
@@ -44,8 +49,15 @@
     - always
   ansible.builtin.import_tasks: cleanup.yml
 
-- name: Gather CRC logs
-  ansible.builtin.import_tasks: crc.yml
-
-- name: Gather EDPM logs
-  ansible.builtin.import_tasks: edpm.yml
+- name: Gather logs when forced or tests failed
+  when: cifmw_artifacts_gather_logs | default('false') | bool or not cifmw_success_flag.stat.exists
+  block:
+    - name: Run os_must_gather
+      ansible.builtin.import_role:
+        name: os_must_gather
+  rescue:
+    - name: Get CRC logs if os_must_gather failed
+      ansible.builtin.import_tasks: crc.yml
+  always:
```

Comment:
IIUC @cjeanner you want to skip *any* logs collection on successful run.
I created a block around `os_must_gather` and `crc.yml` with `edpm.yml` . I hope it's going to allow us to spare a disk space and time execution as well.

Reply:
we probably don't need the `always` block for the edpm.yml inclusion.

But that's more cosmetics at this point. Should do the trick.

Reply:
hmm yep, by adding the same `when` condition :).